bin/roundtrip-translation.sh

```diff
@@ -0,0 +1,393 @@
+
+  if [ "${VERBOSE}" = true ]; then 
+    echo -e "${YELLOW_FONT}WARN: ${ERRMSG}" >&2
+    echo -e "${DEFAULT_COLOR}"
+  fi
+}
+
+function warnDefault() {
+    VALUE=$1
+    MESSAGE=$2
+
+    if [ "${VERBOSE}" = true -a  -z "${VALUE}" ]; then 
+        echo -e "${YELLOW_FONT}WARN: ${MESSAGE}" >&2
+        echo -e "${DEFAULT_COLOR}"
+    fi
+}
+
+
+# Process command line args
+while [[ $# -gt 0 ]]; do
+  opt="$1"
+  shift;
+
+  case "$opt" in
+    "-h"|"--help"            ) help; exit 0;;
+    "-d"|"--data"            ) JSONLD_DATA_FILE="$1"; shift;;
+    "-b"|"--backtarget"      ) BACK_TARGET_TRANSLATION_FILE="$1"; 
+                               check_required_arg "${BACK_TARGET_TRANSLATION_FILE}" "ERROR: ${opt} specified with no back target translation file!"
+                               verify_file_exists "${BACK_TARGET_TRANSLATION_FILE}" "shex"
+                               shift;;
+    "-f"|"--frame"           ) FRAME_FILE="$1"; 
+                               check_required_arg "${FRAME_FILE}" "ERROR: ${opt} specified with no frame file!"
+                               verify_file_exists "${FRAME_FILE}" "frame"
+                               shift;;
+    "-j"|"--jsonvars"        ) JSON_VARS_FILE="$1"; 
+                               check_required_arg "${JSON_VARS_FILE}" "ERROR: ${opt} specified with no JSON variables file!"
+                               verify_file_exists "${JSON_VARS_FILE}" "json"
+                               shift;;
+    "-o"|"--output"          ) WORK_DIR="$1"; 
+                               check_required_arg "${WORK_DIR}" "ERROR: ${opt} specified with no output directory!"
+                               shift;;
+    "-r"|"--root"            ) RDF_ROOT="$1"; 
+                               error_exit "${RDF_ROOT}" "${opt} specified with no RDF root!"
+                               shift;;
+    "-s"|"--source"          ) SOURCE_DATA_SCHEMA_FILE="$1"; 
+                               error_exit "${SOURCE_DATA_SCHEMA_FILE}" "${opt} specified with no source ShEx schema file!"
+                               shift;;
+    "-t"|"--target"          ) SHEX_TARGET_TRANSLATION_FILE="$1"; shift;;
+    "-v"|"--verbose"         ) VERBOSE=true;; 
+    *                        ) echo "ERROR: Invalid option: \""$opt"\"" >&2; usage; exit 1;;
+    esac
+done
+
+check_required_arg "${JSONLD_DATA_FILE}" "ERROR: expected a JSON-LD data file!" 
+verify_file_exists "${JSONLD_DATA_FILE}" "jsonld" 
+
+check_required_arg "${SHEX_TARGET_TRANSLATION_FILE}" "ERROR: expected a ShEx translation file!" 
+verify_file_exists "${SHEX_TARGET_TRANSLATION_FILE}" "shex" 
+
+WORK_DIR=${WORK_DIR:-./work}
+
+# Get the RDF Root.  If we are using hokukahu as our root, set SED files to true.
+# This will do editing on the sed files to ensure a smooth round trip with the current ShEx requirements
+warnDefault "${RDF_ROOT}" "No RDF root specified; defaulting to http://hokukahu.com/patient-1."
+RDF_ROOT=${RDF_ROOT:-http://hokukahu.com/patient-1}
```

Comment:
Unless you know of some reason why we shouldn't, let's change from http://hokukahu.com/patient-1 to urn:local:patient-1 throughout, since that is a made-up URI that may change and is not dereferenceable.